contrib/fedora/rpm/build.sh

```diff
@@ -0,0 +1,329 @@
+
+TARBALL_NAME="firewalld-$VERSION.tar.bz2"
+CALL_DIST=0
+
+if [ "$TARBALL" = "" -a "$ARG_FORCE" != 1 -a -f "./$TARBALL_NAME" ]; then
+    TARBALL="./$TARBALL_NAME"
+fi
+
+if [ "$TARBALL" != "" ]; then
+    :
+else
+    if [ "$ARG_FORCE" = 1 ]; then
+        git clean -fdx -- ":(exclude)$RPMDIR/" ":(exclude)firewalld-*.tar.bz"
+    else
+        if [ "$(git clean -ndx -- ":(exclude)$RPMDIR/" ":(exclude)firewalld-*.tar.bz")" != "" ] ; then
+            die "Working directory is not clean. Rerun with --force. WARNING: this cleans the working directory!"
+        fi
+    fi
+    CALL_DIST=1
+fi
+
+BUILDDIR="$(mktemp -d -p "$RPMDIR" "Builddir.$DATE.XXXXX")"
+
+LOG "conf: BUILDDIR=\"$BUILDDIR\""
+
+ln -snf "$(basename "$BUILDDIR")" "$RPMDIR/Builddir.building"
+mkdir -p "$BUILDDIR/SPECS/"
+mkdir -p "$BUILDDIR/SOURCES/"
+
+ln "$BUILDLOG" "$BUILDDIR/build.log"
+rm -f "$BUILDLOG"
+
+SPECFILE="$BUILDDIR/SPECS/firewalld.spec"
+
+dist() {
+    ./autogen.sh
+    ./configure
+    make dist
+    TARBALL="./$TARBALL_NAME"
+}
+
+if [ "$CALL_DIST" = "1" ]; then
+    dist
+fi
+
+if [ ! -f "$TARBALL" -o "$(printf "%s" "$TARBALL" | wc -w)" -ne 1 ] ; then
+    if [ "$CALL_DIST" = "1" ]; then
+        die "Tarball not generated by \`make dist\`. Something went wrong."
+    fi
+    die "To tarball found. Build with --force (warning: cleans git tree)"
+fi
+
+write_changelog() {
+    if [ -z "$CHANGELOG" ]; then
+        cat <<- EOF
+	* $(LC_TIME=C date '+%a %b %d %Y') $FW_BUILD_USERNAME - %{epoch_version}:%{version}-%{release_version}%{?snap}
+	- build of firewalld ($DATE, uuid: $UUID, git: $COMMIT_FULL)
+	$(git log -n20 --date=local --format='- %h %s [%an] (%ci)')
+	- ...
+	EOF
+    else
+        echo "$CHANGELOG"
+    fi > "$BUILDDIR/SOURCES/CHANGELOG"
+}
+
+write_changelog
+
+cat "$RPMDIR/firewalld.spec" \
+  | sed \
+        -e "s/__VERSION__/$VERSION/g" \
+        -e "s/__RELEASE_VERSION__/$RELEASE_VERSION/g" \
+        -e "s/__SNAPSHOT__/$FW_BUILD_SNAPSHOT/g" \
+        -e "s/__COMMIT__/$COMMIT/g" \
+        -e "s/__COMMIT_FULL__/$COMMIT_FULL/g" \
+  | sed -e "/^__CHANGELOG__$/ \
+        {
+            r $BUILDDIR/SOURCES/CHANGELOG
+            d
+        }" > "$SPECFILE" || die "Error generating spec file \"$SPECFILE\""
+
+for f in \
+    contrib/fedora/rpm/1001-fedora-only-MDNS-default.patch \
+    contrib/fedora/rpm/2001-RHEL-only-Add-cockpit-by-default-to-some-zones.patch \
+    contrib/fedora/rpm/FedoraServer.xml \
+    contrib/fedora/rpm/FedoraWorkstation.xml \
+    contrib/fedora/rpm/org.fedoraproject.FirewallD1.desktop.rules.choice \
+    "$TARBALL" \
+    ; do
+    cp "$f" "$BUILDDIR/SOURCES/" || die "Could not copy file \"$f\" to \"$BUILDDIR/SOURCES/\""
+done
+
+if [ "$BUILDTYPE" != "sources" ]; then
+    if [ "$BUILDTYPE" = "srpm" ]; then
+        RPM_BUILD_OPTION=-bs
+    else
+        RPM_BUILD_OPTION=-ba
+    fi
+
+    list_add_arg ARGS_WITH_X    "--with"    "${ARGS_WITH[@]}"
+    list_add_arg ARGS_WITHOUT_X "--without" "${ARGS_WITHOUT[@]}"
+
```

Comment:
Do we anticipate building a `fedora_variant` on a non-fedora OS? If then answer is "no", then we can omit all this with logic entirely as it's detected in the spec file itself.

Reply:
the main reason why `%bcond_with fedora_variant` exists, is so that the various places have a check
```
%if %{with fedora_variant}
```
instead of
```
%if 0%{?fedora}
```

That means, the value of this variable is that you can grep for `fedora_variant`, and find all places related to that.

At that point, making it a bcond variable is a small extension. Yes, it allows to build a fedora-variant on RHEL and vice versa. Nice.

---

More interesting is `firewalld_test`. Maybe `firewalld_test` package should *always* be build (even on RHEL), and we should exclude the `firewalld-test` via other means from installation. But as it is for now, whether that package gets build is a bcond switch. It would seem a loss, if I cannot build a `firewalld-test` package on RHEL.